File: workshop_content/KI_WORKSHOP/C#_Vergleichsgruppe/Provisionierung/Domain/SystemProvisionContext.cs
namespace Provisionierung.Domain;

/// <summary>
/// Standard-Implementierung basierend auf Systemzeit.
/// </summary>
public class SystemProvisionContext : IProvisionContext
{
    public DateTime AktuellesDatum => DateTime.Now;

    public bool IstFreitagNach15Uhr()
    {
        var now = AktuellesDatum;
        return now.DayOfWeek == DayOfWeek.Friday && now.Hour >= 15;
    }
}

File: workshop_content/KI_WORKSHOP/Java_Vergleichsgruppe/src/main/java/provisionierung/domain/SystemProvisionContext.java
package provisionierung.domain;

import java.time.LocalDateTime;

/**
 * Standard-Implementierung basierend auf Systemzeit.
 */
public class SystemProvisionContext implements ProvisionContext {
    @Override
    public LocalDateTime getAktuellesDatum() {
        return LocalDateTime.now();
    }
}
